kumascript/macros/CSSSyntax.ejs

```diff
@@ -165,14 +165,22 @@ function getPropertySyntax(propertyName) {
   if (multiplierName.startsWith('{')) {
     key = '{}';
   }
-  // these two multiplier combinations can appear, and in this case we want links to both parts
+  // these two multiplier combinations can appear, we want to annotate them separately
   if (multiplierName === '+#' || multiplierName === '#?') {
     const info1 =  syntaxDescriptions[multiplierName[0]];
     const info2 =  syntaxDescriptions[multiplierName[1]];
     const link1 = `<a href="${valueDefinitionUrl}#${info1.fragment}" title="${info1.tooltip}">${multiplierName[0]}</a>`;
     const link2 = `<a href="${valueDefinitionUrl}#${info2.fragment}" title="${info2.tooltip}">${multiplierName[1]}</a>`;
     return `${link1}${link2}`;
   }
+  // the "#" multipler can be followed by the curly brackets, we want to annotate them separately
+  if (multiplierName.startsWith("#{")) {
+    const info1 =  syntaxDescriptions[multiplierName[0]];
+    const info2 =  syntaxDescriptions["{}"];
+    const link1 = `<a href="${valueDefinitionUrl}#${info1.fragment}" title="${info1.tooltip}">${multiplierName[0]}</a>`;
+    const link2 = `<a href="${valueDefinitionUrl}#${info2.fragment}" title="${info2.tooltip}">${multiplierName.slice(1)}</a>`;
+    return `${link1}${link2}`;
```

Comment:
```suggestion
    const info = syntaxDescriptions["#{"];
    return `<a href="${valueDefinitionUrl}#${info.fragment}" title="${info.tooltip}">${multiplierName}</a>`;
```

The syntax `{3}` and `#{3}` are different and we shouldn't split `#{` for the ref URLs. :raised_eyebrow:
`value{3}` => `value value value`
`value#{3}` => `value, value, value`
The `curly_braces` fragment takes to [the wrong section](https://developer.mozilla.org/en-US/docs/Web/CSS/Value_definition_syntax#curly_braces) in this scenario.

We either update exiting `#` entry in the `syntaxDescriptions` or create a new one for `#{` (e.g. hash_mark_with_range).
Also, the info needs to be updated at content side.

Reply:
@wbamberg What do you think about Onkar's concerns?

Reply:
@OnkarRuikar The `#` entry in `syntaxDescriptions` currently states:

> Hash mark: the entity is repeated one or several times, each occurence separated by a comma

In what way does it need to be updated? Wouldn't it be sufficient to update the [Curly braces ({ })](https://developer.mozilla.org/en-US/docs/Web/CSS/Value_definition_syntax#curly_braces) section to mention that the values are separated by space, except if the curly braces are preceded by a [Hash mark (#)](https://developer.mozilla.org/en-US/docs/Web/CSS/Value_definition_syntax#hash_mark), in which case they are separated by a comma?

Reply:
@caugner my main concern is that `#{ }` needs to be presented as one link (as one modifier). And we shouldn't split it.

> Wouldn't it be sufficient to update the [Curly braces ({ })](https://developer.mozilla.org/en-US/docs/Web/CSS/Value_definition_syntax#curly_braces) section...

From the implementation angle your point makes things easy.

But grouping the modifier descriptions based on presence of comma is easy to understand. Like the [spec](https://web.archive.org/web/20220424210922/https://drafts.csswg.org/css-values/#component-multipliers) has done by putting `{ }` and `#{ }` in separate points.

Amending Curly braces section may complicate the things for readers. Plus separate examples will also be required in the same section. Examples without hash and examples with hash.
But mentioning the extra range construct in Hash mark section will be easy to grasp. And the existing examples will be sufficient.

Reply:
So basically, we should a new fragment, _hash_and_brace_, with a tooltip, then replace L178-182 with `key = hash_and_brace`?

Reply:
> So basically, we should a new fragment, _hash_and_brace_, with a tooltip, then replace L178-182 with `key = hash_and_brace`?

Yes. That would be perfect as it'll keep the sections short and simple.

Reply:
> Like the [spec](https://web.archive.org/web/20220424210922/https://drafts.csswg.org/css-values/#component-multipliers) has done by putting `{ }` and `#{ }` in separate points.

@OnkarRuikar Just like on MDN, the spec has separate points for curly braces (`{ }`) and hash mark (`#`), but no separate point for `#{ }`:

<img width="786" alt="image" src="https://user-images.githubusercontent.com/495429/188908451-99e68d21-f894-453d-9f43-13e01882f51e.png">

Reply:
Sorry to be slow replying. I don't think we need to do this. The spec doesn't call out `#{}` as a distinct notation - it actually references the `{}` notation, as in the quote at https://github.com/mdn/yari/pull/7049#discussion_r964945962.

I don't mind if someone else wants to file (and review) a PR to change the way we annotate formal syntax, but I don't propose to do it myself.